Gifski/MainWindowController.swift

```diff
@@ -65,16 +89,23 @@ final class MainWindowController: NSWindowController {
 			$0.makeVibrant()
 		}
 
+		view?.addSubview(cancelButton)
 		view?.addSubview(circularProgress)
-		view?.addSubview(videoDropView, positioned: .above, relativeTo: nil)
 		view?.addSubview(showInFinderButton)
+		view?.addSubview(hoverView)
+		view?.addSubview(videoDropView, positioned: .above, relativeTo: nil)
 
 		window.makeKeyAndOrderFront(nil)
 		NSApp.activate(ignoringOtherApps: false)
 
 		DockProgress.style = .circle(radius: 55, color: .appTheme)
 	}
 
+	@objc
+	func cancel(_ sender: Any?) {
```

Comment:
This could use a comment about where it's coming from. I could also not find this in the docs? Could you?

Reply:
The only mention of `cancel(_:)` is in `NSSavePanel`, which does not seem immediately related. I don't really like that's it's undocumented.

The alternatives are implementing `cancelOperation(_:)` in a `NSWindow` or `NSView` subclass. The latter will probably require jumping through some extra hoops, accepting first responder etc.

Or falling back to the event listener I implemented first.

Personally, I think there's up and downsides to all of these options. The current implementation seems the most light weight and has my preference.